### PR TITLE
Adding typography styles token

### PR DESCRIPTION
**Pull request by lastquestion**
<del>1. Create `h1.token.yml` file that expose `size, color, weight, line-height`. 
2. These get generated into full `h1.token.cssmodules.css` (and common.js, and SCSS).
   To do this, I edited the `theo-cli` to support directories.
3.Add a `h1.type.cssmodules.css`, which composes the `h1.token.cssmodules.css` (which is autogenerated) into a `.text-syle` which can then be used like this:
   ```
   .restaurant-title-text {
     composes: textStyle from '....h1.type.cssmodules.css';
   }
   ```
<del>I'm not married to anything here, see related issue #26 for some context & motivation. So, to add a new text-style, you create a new `whatever.token.yml`, and a `whatever.type.cssmodules.css`.</del>

<del>I think long term it would be super interesting to consider have theo understand this kind of groupings or file formats and have it generate a special format. I played around with this and can see how it would be done, but I'm not sure if that's the right direction to explore yet.</del>

Add a new design token NPM module called `otkit-typography-styles`. Each of the styles express 4 values:
* `size`
*  `weight`
* `line-height`
* `color`

All of which can be used by each platform appropriately.

Each style value is kebab cased, so for example `subheading`'s `weight` is `subheading-weight`.

### CODE REVIEW

**Comment by nickbalestra** [comment]
I see what you doing here and I can understand your intent.

Is important that we approach tokens in very caution ways and resist yet to the temptation of making those "higher order token" to better exploit some feature of a specific format (in this pr proosal css-modules `compose`). As you pointed out in #26 , such composition will be more appropriate to exist  in the consumer-land where you'll be free to build such classes out of tokens and recompose them, and actually will totally make sense there. The less opinionated tokens will be the more successful their adoption.

Some other thoughts:
- This could makes the token system too convolute and complex on two side: consumer-land: with a more fragmented API and "sub" tokens to imports and here on the producer-land side with a less obvious structure (we are aiming for max contribution herw, therefore we should make an effort to reduce barrier not viceversa). 
- Could drive the token project more toward a specific format.
- The very same result could be achieved at consumer site, where I can totally see you build those groups and classes to be composed, so yes, this feel definitely more at home in a repo like the ot-react-ui and therefore not really needed here.

**Comment by lastquestion** [comment]
I'll write more later, but yes, I want to make sure that code doesn't matter here. It's a good way to start the conversation. When doing this work, I was more interested in starting some thinking then anything else.

**Comment by lastquestion** [comment]
Hm, after some time thinking, here are some of my thoughts about this.

* I think building "the actual implementation of the token" in the point of consumption seems valid. I want to think about that problem because it pushes down consistency and re-implementation issues further down the stack. It's a balance though. I think CSS and SCSS is just "too easy" for people to slip into more and more complicated rules, e.g. implementation. And I think that's what I also fell into too. As an analogy, we'd never export an Java .JAR class to correctly construct fonts in this repo, so why we would do that for CSS?

* But, I believe that "H1 text style", or "H2 on dark background", or "H2 on light background", or "subtext", are themselves tokens, and I want to think about how to encode those tokens. This would not be the implementation of those tokens - but, I would like to encode an abstract grouping of simpler tokens. The way I did this here was to generate additional files that depict the "values" of the "token", in a platform agnostic way. I think this part is good and worth further discussion. I considered using a CSS string format as well, as it's easy to parse by multiple platforms. But really, JSON is the better interchange format here. So I kind of like the way I did it here by building multiple files.

**Comment by lastquestion** [comment]
^- I removed the fancy CSS rule file

**Comment by lastquestion** [comment]
I simplified the problem by making a new NPM module, this removes the Theo CLI changes, and then I just use kebab case to namespace the values into groups. Simple and effective.

This has most of the styles, down to subheading, but it's missing `body` and `subtext` (and variants of that)

File: OTKit/otkit-typography-styles/token.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    value: "{!font-weight-bold}"
+  h2-line-height:
+    value: "{!line-height-large}"
+
+  h3-size:
+    value: "{!font-size-medium}"
+  h3-color:
+    value: "{!color-gray-primary}"
+  h3-weight:
+    value: "{!font-weight-normal}"
```
Comment (lastquestion) [inline OTKit/otkit-typography-styles/token.yml]: `h3` and `h2` used weight `normal` and `medium` interchangeably in the invision board. I'm pretty sure you meant `normal` for both. cc @mannionaco @francesyun @lindseyhight

Comment (nickbalestra) [inline OTKit/otkit-typography-styles/token.yml]: Now that you mention this, I noted you [used](https://github.com/opentable/design-tokens/blob/master/OTKit/aliases.yml#L81) 'normal' and 'bold' what that does it means? I think that they should have an actual weight value I guess @lastquestion ?

Comment (karuto) [inline OTKit/otkit-typography-styles/token.yml]: @nickbalestra The font weight "normal" = 400 and "bold" = 700. I agree that these can be changed into actual numerical weight values as they have no feedback. 

The only special gotcha in CSS font weight is the fallback which only applies for numeric 400 and 500 values, so really "normal" and "bold" is the same as 400 and 700 here.

Comment (nickbalestra) [inline OTKit/otkit-typography-styles/token.yml]: Yup, I'm trying to think in general terms here and not too tight with CSS :)

**Comment by lastquestion** [comment]
I like the name of headings also. Let's change that.

About having a separate token repo: I thought you wanted that. Let me fix that too. :)

**Comment by lastquestion** [comment]
I'll reopen this with a rebased commit.

**Pull request closed without merging**